docs: add post about certificates (#1104)
kedacore/keda-docs: content/blog/2023-04-20-certificate-improvements.md

+++

> title = "Exploring the new certificate management in KEDA 2.10: Enhancing security for autoscaling"

Reviewer: Just a thought:
```suggestion
title = "Securing autoscaling with the newly improved certificate management in KEDA 2.10"
```

date = 2023-04-20
author = "Jorge Turrado (SCRM Lidl International Hub)"
aliases = [
"/blog/exploring-new-certificate-management"
]
+++

> Recently we have release KEDA v2.10, which has come with some important improvements related with the security and certificate management: 

Reviewer: ```suggestion
Recently, we have release KEDA v2.10 that introduced a key improvement for managing certificates and your autoscaling security: 
```

> - The usage of certificates to encrypt any communication between components.

Reviewer: ```suggestion
- Encryption of any communication between KEDA components.
```

> - The support for providing your own certificated for internal communications.

Reviewer: ```suggestion
- Support for providing your own certificates for internal communications.
```

> - The support for providing customs CAs and trust in them.

Reviewer: ```suggestion
- Support for using customs certificate authorities (CA).
```

What do you mean with this?
>  and trust in them

> Thanks to them, we can improve the security between KEDA components, but also between k8s API server and between KEDA and upstreams. Let's explain all of them deeper :) 

Reviewer: Upstreams is, what? The scaler sources?
```suggestion
With these new improvements, we can dramatically improve the security between KEDA components, the Kubernetes API server and scaler sources. Let's take a closer look.
```

```yaml
certificates:
  autoGenerated: false # disables the cert-generation by the operator
  certManager:
    enabled: true # enables the certificate generation using cert manager
    generateCA: true # uses a self-signed generated CA
    # caSecretName: "kedaorg-ca"
```

To provide a CA to be used for generating the certificates, you could save it as a secret (in the same namespace where KEDA is) and provide the secret. This will register an issuer based on it for generating the certificates.

```yaml
certificates:
  autoGenerated: false # disables the cert-generation by the operator
  certManager:
    enabled: true # enables the certificate generation using cert manager
    generateCA: false # uses a self-signed generated CA
    caSecretName: "kedaorg-ca" # secret name where the CA is stored
```

> > Note: As KEDA needs that the services are correctly patched, when the cert manager integration is enabled the manifests are annotated for it. If you provide a CA via secret, the secret has to be annotated with `cert-manager.io/allow-direct-injection: "true"`. Otherwise cert manager won't be able to patch the resources with the CA.

Reviewer: Let's add an outro and links to further details
